apiserver/facades/controller/externalcontrollerupdater/package_test.go

```diff
@@ -9,6 +9,8 @@ import (
 	gc "gopkg.in/check.v1"
 )
 
+//go:generate go run github.com/golang/mock/mockgen -package mocks -destination mocks/domain_mock.go github.com/juju/juju/apiserver/facades/controller/externalcontrollerupdater EcService
```

Comment:
I prefer to generate these into something like _domain_mock_test.go_ in the root package. Everyone keeps creating new packages for mocks, when it was only ever done initially for a specific case to avoid import cycles.